src/corehost/cli/nethost/nethost.cpp

```diff
@@ -21,25 +21,35 @@ namespace
 NETHOST_API int NETHOST_CALLTYPE get_hostfxr_path(
     char_t * buffer,
     size_t * buffer_size,
-    const char_t * assembly_path)
+    const struct get_hostfxr_parameters *parameters)
 {
     if (buffer_size == nullptr)
         return StatusCode::InvalidArgFailure;
 
     trace::setup();
     error_writer_scope_t writer_scope(swallow_trace);
 
-    pal::string_t root_path;
-    if (assembly_path != nullptr)
-        root_path = get_directory(assembly_path);
-
     pal::dll_t fxr;
     pal::string_t fxr_path;
     if (!fxr_resolver::try_get_existing_fxr(&fxr, &fxr_path))
     {
-        pal::string_t dotnet_root;
-        if(!fxr_resolver::try_get_path(root_path, &dotnet_root, &fxr_path))
-            return StatusCode::CoreHostLibMissingFailure;
+        if (parameters != nullptr && parameters->dotnet_root != nullptr)
```

Comment:
The `version` field value must be validated.

Reply:
Also please add a test for that.

Reply:
Added.